src/components/GitmojiList/index.js

```diff
@@ -29,6 +31,12 @@ const GitmojiList = (props: Props): Element<'div'> => {
       })
     : props.gitmojis
 
+  React.useEffect(() => {
+    if (query.search) {
+      setSearchInput(query.search)
+    }
+  }, [query.search])
+
```

Comment:
To update the router.query I suggest using shallow routing:

https://nextjs.org/docs/routing/shallow-routing

```suggestion
React.useEffect(() => {
  if (router.query.search !== searchInput) {
    router.push(`/?search=${searchInput}`, undefined, { shallow: true })
  }
}, [searchInput])
```

Reply:
Sync the query parameter with the input in commit e8e19740b6ba55092982c0e6d23868d9d82dc6f5.

Reply:
I would check the searchInput first, to not have to simplify the destructuring of the object

Reply:
This ensures that if the search input is empty, the param is removed altogether.